### ISSUE TITLE
Building documentation locally fails on MacOS

### ISSUE THREAD

**Issue by prhomhyse** [netdata/netdata#6787]
##### Bug report summary
Following the [contributing documentation](https://github.com/netdata/netdata/blob/master/docs/contributing/contributing-documentation.md) guide. 
I try to run the bash script `sh docs/generator/buildhtml.sh`

I get the below:

```
Preparing directories
find: -printf: unknown primary or operator
Preparing source for en
Creating en mkdocs.yaml
find: -printf: unknown primary or operator
find: -printf: unknown primary or operator
find: -printf: unknown primary or operator
Fixing links
sed: 1: "./packaging/docker/READ ...": invalid command code .
. . .
sed: 1: "./README.md": invalid command code .
. . . 
sed: 1: "./README.md": invalid command code .
sed: 1: "./daemon/config/README.md": invalid command code .
sed: 1: "./DOCUMENTATION.md": invalid command code .
sed: 1: "./SECURITY.md": invalid command code .
/Users/prhomhyse/Lab/netdata/netdata
Calling mkdocs
INFO    -  Cleaning site directory 
INFO    -  Building documentation to directory: /Users/prhomhyse/Lab/netdata/netdata/docs/generator/build 
Traceback (most recent call last):
  File "/usr/local/bin/mkdocs", line 10, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/mkdocs/__main__.py", line 163, in build_command
    ), dirty=not clean)
  File "/usr/local/lib/python3.7/site-packages/mkdocs/commands/build.py", line 267, in build
    nav = get_navigation(files, config)
  File "/usr/local/lib/python3.7/site-packages/mkdocs/structure/nav.py", line 104, in get_navigation
    items = _data_to_navigation(nav_config, files, config)
  File "/usr/local/lib/python3.7/site-packages/mkdocs/structure/nav.py", line 162, in _data_to_navigation
    for item in data
  File "/usr/local/lib/python3.7/site-packages/mkdocs/structure/nav.py", line 162, in <listcomp>
    for item in data
  File "/usr/local/lib/python3.7/site-packages/mkdocs/structure/nav.py", line 155, in _data_to_navigation
    for key, value in data.items()
  File "/usr/local/lib/python3.7/site-packages/mkdocs/structure/nav.py", line 155, in <listcomp>
    for key, value in data.items()
  File "/usr/local/lib/python3.7/site-packages/mkdocs/structure/nav.py", line 165, in _data_to_navigation
    file = files.get_file_from_path(path)
  File "/usr/local/lib/python3.7/site-packages/mkdocs/structure/files.py", line 33, in get_file_from_path
    return self.src_paths.get(os.path.normpath(path))
  File "/usr/local/Cellar/python/3.7.4/Frameworks/Python.framework/Versions/3.7/lib/python3.7/posixpath.py", line 340, in normpath
    path = os.fspath(path)
TypeError: expected str, bytes or os.PathLike object, not NoneType

```
The full output can be found in this [Gist](https://gist.github.com/prhomhyse/f89778697b4eb4153e077b92bc5bcd39)

In essence, the MkDocs site fails to build. 

Trying again to run `mkdocs build`

```
mkdocs build
INFO    -  Cleaning site directory 
INFO    -  Building documentation to directory: /Users/prhomhyse/Lab/netdata/netdata/docs/generator/build 
Traceback (most recent call last):
  File "/usr/local/bin/mkdocs", line 10, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/mkdocs/__main__.py", line 163, in build_command
    ), dirty=not clean)
  File "/usr/local/lib/python3.7/site-packages/mkdocs/commands/build.py", line 267, in build
    nav = get_navigation(files, config)
  File "/usr/local/lib/python3.7/site-packages/mkdocs/structure/nav.py", line 104, in get_navigation
    items = _data_to_navigation(nav_config, files, config)
  File "/usr/local/lib/python3.7/site-packages/mkdocs/structure/nav.py", line 162, in _data_to_navigation
    for item in data
  File "/usr/local/lib/python3.7/site-packages/mkdocs/structure/nav.py", line 162, in <listcomp>
    for item in data
  File "/usr/local/lib/python3.7/site-packages/mkdocs/structure/nav.py", line 155, in _data_to_navigation
    for key, value in data.items()
  File "/usr/local/lib/python3.7/site-packages/mkdocs/structure/nav.py", line 155, in <listcomp>
    for key, value in data.items()
  File "/usr/local/lib/python3.7/site-packages/mkdocs/structure/nav.py", line 165, in _data_to_navigation
    file = files.get_file_from_path(path)
  File "/usr/local/lib/python3.7/site-packages/mkdocs/structure/files.py", line 33, in get_file_from_path
    return self.src_paths.get(os.path.normpath(path))
  File "/usr/local/Cellar/python/3.7.4/Frameworks/Python.framework/Versions/3.7/lib/python3.7/posixpath.py", line 340, in normpath
    path = os.fspath(path)
TypeError: expected str, bytes or os.PathLike object, not NoneType
```

MkDocs works fine for other documentation sites I generated using `mkdocs new` or random ones I cloned to test with `mkdocs build && mkdocs serve`

##### OS / Environment
MacOS
##### Component Name
Docs
##### Steps To Reproduce
Simply run this script
`sh docs/generator/buildhtml.sh`
##### Expected behavior
`sh docs/generator/buildhtml.sh` should build the documentation site locally. 
This site will be located at `docs/generator/build/`.

**Comment by cakrit** [comment]
We didn't intend for this to run everywhere, but leaving it open, in case someone wants to do a PR to make it compatible for MacOS.

**Comment by prhomhyse** [comment]
> We didn't intend for this to run everywhere

This makes sense. 
It was frustrating trying to make it run.
Since the priority is low, I can take a stab at it much later.

**Comment by paulkatsoulakis** [comment]
It's a BASH script and you are using `sh` to execute it.

Try the following commands in sequence:
1) `chmod +x docs/generator/buildhtml.sh`
2) `./docs/generator/buildhtml.sh`

and let us know if that works

**Comment by prhomhyse** [comment]
@paulkatsoulakis 

It doesn't work. 

I tried the commands in sequence and the output is still the same as the [Gist](https://gist.github.com/prhomhyse/f89778697b4eb4153e077b92bc5bcd39)

**Comment by jurjendijkstra** [comment]
> We didn't intend for this to run everywhere, but leaving it open, in case someone wants to do a PR to make it compatible for MacOS.

I suffer the same error but I am not on MacOS.  I don't know any details yet (am still investigating), just wanted to say it is not MacOS specific.

**Comment by jurjendijkstra** [comment]
I found that the error happens when the "nav:" in mkdocs.yml contains a header that has no children.
For example:
`            - 'Masterdata':
            - 'Project':
               - 'New': 'windmaakprjxe.md'
`
That line " - 'Masterdata':"  is the culprit because it has no child items.

**Comment by prhomhyse** [comment]
Hi @jurgenhaas 
> I suffer the same error but I am not on MacOS. I don't know any details yet (am still investigating), just wanted to say it is not MacOS specific.

What OS did you build the documentation locally on?

**Comment by jurjendijkstra** [comment]
jurgenhaas?  :-)  OS is a bit unclear: a colleague has put mkdocs in a Docker container. I myself do not have access to the Dockerfile, all I know is that it is based on some linux. I am using that Docker on my Windows 10 PC. But I believe it is irrelevant for this issue which OS it is, because the root issue is that the nav: section in mkdocs.yml was invalid and it would have helped tremendously if mkdocs had a nice validation saying what the problem was, instead of this low-level Type error from posixpath.py.

**Comment by joelhans** [comment]
This was actually fixed by #7688. Closing!
